docs/concepts.rst

```diff
@@ -0,0 +1,154 @@
+are mutable, which means new facts can be inserted to the table, and
+existing facts can be updated or deleted. Tables can be created from a
+Kafka topic or derived from existing streams and tables. In both cases,
+a table’s underlying data is durably stored (persisted) within a Kafka
+topic on the Kafka brokers.
+
+.. _modes-of-operation:
+
+==================
+Modes of operation
+==================
+
+Standalone mode
+---------------
+
+In stand-alone mode, both the KSQL client and server components are
+co-located on the same machine, in the same JVM, and are started
+together. This makes standalone mode very convenient for local
+development and testing.
+
+.. image:: img/standalone-mode.png
+
+To run KSQL in standalone mode:
+
+-  Start the KSQL CLI and the server components all in the same JVM:
+
+   -  Start with default settings:
+
+      .. code:: bash
+
+		$ ./bin/ksql-cli local
+
+   -  Start with :ref:`custom
+      settings <configuring-ksql>`, pointing
+      KSQL at a specific Kafka cluster (see the Streams
+      `bootstrap.servers <streams_developer-guide_required-configs>`
+      setting):
+
+      .. code:: bash
+
+		$ ./bin/ksql-cli local --bootstrap-server kafka-broker-1:9092 \
+                       --properties-file path/to/ksql-cli.properties
+
+Client-server mode
+------------------
+
+In client-server mode, you can run a pool of KSQL servers on remote
+machines, VMs, or containers. The CLI then connects to these remote KSQL
+servers over HTTP.
+
+.. image:: img/client-server.png
+
+To run KSQL in client-server mode:
+
+-  Start any number of server nodes:
+
+   -  Start with default settings:
+
+      .. code:: bash
+
+		$ ./bin/ksql-server-start
+
+   -  Start with :ref:`custom
+      settings <configuring-ksql>`, pointing
+      KSQL at a specific Kafka cluster (see Streams :ref:`bootstrap servers <streams_developer-guide_required-configs>` setting):
+
+      .. code:: bash
+
+         $ hostname
+         my-ksql-server
+
+         $ cat ksql-server.properties
+         # You must set at least the following two properties
+         bootstrap.servers=kafka-broker-1:9092
+         # Note: `application.id` is not really needed but you must set it
+         #       because of a known issue in the KSQL Developer Preview
+         application.id=app-id-setting-is-ignored
+         
+         # Optional settings below, only for illustration purposes
+         # The hostname/port on which the server node will listen for client connections
+         listeners=http://0.0.0.0:8090
+
+      To start the server node with the settings above:
+
+      .. code:: bash
+
+		$ ./bin/ksql-server-start ksql-server.properties
+
+-  Start any number of CLIs, specifying the desired KSQL server address
+   as the ``remote`` endpoint:
+
+   .. code:: bash
+
+       $ ./bin/ksql-cli remote http://my-ksql-server:8090
+
+All KSQL servers (and their engines) share the work of processing KSQL
```

Comment:
@bluemonk3y : We should update this section on how to actually pool the servers.

Reply:
A KSQL server pool is one where all servers share the same 'command' topic. The default command topic is named: 'ksql__commands'. The command topic name can be changed by applying the following system properties to the ksqlserver.properties file. 
ksql.service.id = ksql_ 
ksql.command.topic.suffix=commands

Reply:
Good draft! Here's my slightly modified one:

> KSQL servers that share the same "command" topic belong to the same resource pool.  By default, a KSQL server uses the command topic named `ksql__commands`.  To assign a server to a different pool than the default, change the `ksql.command.topic.suffix` setting in its configuration:
>
> ```properties
> # Default value: `commands`.
> #
> # Changing this to `production_commands` as shown below will result in
> # the command topic named `ksql__production_commands` being used.
> ksql.command.topic.suffix = production_commands
> ```